src/DataCollectors/TraceDataCollector/VanguardCollector/DynamicCoverageDataCollector.cs

```diff
@@ -0,0 +1,125 @@
+        {
+        }
+
+        internal DynamicCoverageDataCollector(
+            ICollectorUtility collectorUtility,
+            IDynamicCoverageDataCollectorImpl dynamicCoverageDataCollectorImpl)
+        {
+            this.collectorUtility = collectorUtility;
+            this.implementation = dynamicCoverageDataCollectorImpl;
+        }
+
+        protected override void OnInitialize(XmlElement configurationElement)
+        {
+            this.implementation.Initialize(configurationElement, this.DataSink, this.Logger);
+            this.Events.SessionStart += this.SessionStart;
+            this.Events.SessionEnd += this.SessionEnd;
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        /// <param name="disposing"> The disposing</param>
+        protected override void Dispose(bool disposing)
+        {
+            this.Events.SessionStart -= this.SessionStart;
+            this.Events.SessionEnd -= this.SessionEnd;
+            this.implementation?.Dispose();
+            base.Dispose(disposing);
+        }
+
+        /// <summary>
+        /// The GetEnvironmentVariables
+        /// </summary>
+        /// <returns>Returns EnvironmentVariables required for code coverage profiler. </returns>
+        protected override IEnumerable<KeyValuePair<string, string>> GetEnvironmentVariables()
+        {
+            var vanguardDirectory = this.collectorUtility.GetVanguardDirectory();
+            var vanguardX86ProfilerFullPath = Path.Combine(vanguardDirectory, VanguardX86ProfilerPath);
+            var vanguardX64ProfilerFullPath = Path.Combine(vanguardDirectory, VanguardX64ProfilerPath);
+            var envVaribles = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>(DynamicCoverageDataCollector.CoreclrEnableProfilingVariable, "1"),
+                new KeyValuePair<string, string>(DynamicCoverageDataCollector.CoreclrProfilerPathVariable32, vanguardX86ProfilerFullPath),
+                new KeyValuePair<string, string>(DynamicCoverageDataCollector.CoreclrProfilerPathVariable64, vanguardX64ProfilerFullPath),
+                new KeyValuePair<string, string>(DynamicCoverageDataCollector.CoreclrProfilerVariable, VanguardProfilerGuid),
+                new KeyValuePair<string, string>(DynamicCoverageDataCollector.CodeCoverageSessionNameVariable, this.implementation.GetSessionName()),
+                new KeyValuePair<string, string>(DynamicCoverageDataCollector.CorEnableProfilingVariable, "1"),
+                new KeyValuePair<string, string>(DynamicCoverageDataCollector.CorProfilerPathVariable32, vanguardX86ProfilerFullPath),
+                new KeyValuePair<string, string>(DynamicCoverageDataCollector.CorProfilerPathVariable64, vanguardX64ProfilerFullPath),
+                new KeyValuePair<string, string>(DynamicCoverageDataCollector.FullCorProfiler, VanguardProfilerGuid),
```

Comment:
>FullCorProfiler [](start = 78, length = 15)

Nit: This stands out, can we rename it to "CLRProfilerVariable"

Reply:
Done.